client/state/themes/actions/request-theme.js

```diff
@@ -101,3 +102,82 @@ export function requestTheme( themeId, siteId, locale ) {
 			} );
 	};
 }
+
+export function requestSingleTheme( themeId, siteId, sourceType, locale ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: THEME_REQUEST,
+			siteId,
+			themeId,
+		} );
+		if ( sourceType === 'wporg' ) {
+			return fetchWporgThemeInformation( themeId )
```

Comment:
We aren't returning anything here, when the promise is resolved we're just dispatching an action. We can remove this return statement.